src/dashboard-refactor/lists-sidebar/components/lists-search-bar/list-search.ts

```diff
@@ -0,0 +1,49 @@
+import { ListsSearchables, ListsSearchResult } from './types'
+
+export const searchLists = (
+    searchStr: string,
+    searchData: ListsSearchables,
+): ListsSearchResult[] => {
+    const { sources, lists } = searchData
+    // 1. Loop through sources and create an array with an object per source
+    //  - This should have a listSource prop and a resultsList prop with empty
+    //    array value
+    return sources.reduce((arr, source) => {
+        arr.push({
+            listSource: source,
+            resultsList: [],
+        })
+        // 2. For each source run through lists and run search func. If match exists
+        //    then add to resultsList array
+        const sourceItemArr = arr[arr.length - 1]
+        lists.map((list) => {
```

Comment:
While this will do what you want here, Array.map is generally used to map each element in an array to something new, returning the new list with all those newly mapped elements. Here you're just using it to loop through the list, so more appropriate would be to use Array.forEach, or standard for..of loop `for (const list of lists) { ... }`

Reply:
Makes sense. I think I knew this, but didn't really think about it in the implementation so it's a good reminder.